pkg/cluster/defaults.go

```diff
@@ -88,3 +107,34 @@ func setMachineHealthCheckTimeoutDefaults(cluster *anywherev1.Cluster, nodeStart
 		}
 	}
 }
+
+// setMachineHealthCheckMaxUnhealthyDefaults sets default maxUnhealthy values for cluster's machine health checks.
+func setMachineHealthCheckMaxUnhealthyDefaults(cluster *anywherev1.Cluster, globalMaxUnhealthy, workerMaxUnhealthy intstr.IntOrString) {
+	topLevelMaxUnhealthyUndefined := true
+	if cluster.Spec.MachineHealthCheck.MaxUnhealthy == nil {
+		cluster.Spec.MachineHealthCheck.MaxUnhealthy = &globalMaxUnhealthy
+	} else {
+		topLevelMaxUnhealthyUndefined = false
+	}
```

Comment:
Here is my thinking - if top-level MHC maxUnhealthy is defined, I think that should override the worker's defaults too. We know that CP default = top-level default = 100%, but worker default = 40%. So if I set CP default and worker defaults, then come to the top-level default and see that it's set, I will have to go back and change all of the worker nodes again to 100%. So to avoid the double processing, I am checking it at the top and setting the flag so I can use it later.

Reply:
This changes if we decide that the top-level setting doesn't affect the worker nodes, but in that case it's only acting as CP Max unhealthy which I thought was confusing.

Reply:
I agree. This differs from how we usually handle modular configurations, but it makes sense for this use case.